packages/amplify_core/lib/src/types/models/async_model.dart

```diff
@@ -31,35 +32,34 @@ class AsyncModel<
   }
 
   T? _cache;
-  FutureOr<T> Function()? _model;
+  FutureOr<T> Function(DataStorePluginInterface)? _model;
+  AsyncMemoizer<T>? _getMemo;
 
   /// Returns the cached model, if any.
   ///
   /// Use [get] to retrieve the model if it is not cached.
-  T? get cache {
-    final cache = _cache;
-    if (cache != null) {
-      return cache;
-    }
-    final model = _model!;
-    if (model is T Function()) {
-      return _save(model());
-    }
-    return null;
-  }
+  T? get cache => _cache;
 
   /// Retrieves the model, returning a [Future] if the model is not available
   /// synchronously.
-  FutureOr<T> get() {
+  FutureOr<T> get([DataStorePluginInterface? dataStore]) {
```

Comment:
This impl and exact interface of this are still in flux - no tests have been written yet. I've mainly added this so that codegen had something to target for now.

The goal of this would be to cache the retrieval of a specific value once. Updates to the underlying data would still be reported through the current mechanisms like `observe` although I'd love to see a mix of the two in a future evolution of this class - but that would likely be a large change + need org-wide review.